### DISCUSSION TITLE
How do I copy/share from my iphone google authentication app to here?

### DISCUSSION THREAD

**Discussion by ndvbd** [Authenticator-Extension/Authenticator#620] [General]
How do I copy/share from my iphone google authentication app to here?

**Comment by mymindstorm** [comment]
We have a feature that allows you to import Google Authenticator QR codes that will be released with the next version.

**Comment by chenglu01** [comment]
> We have a feature that allows you to import Google Authenticator QR codes that will be released with the next version.

 When will the next version be released ？

**Comment by ira-at-work** [comment]
Can't wait :-)
I see the latest version on github is the same 6.2.1 as on the dev branch and no relevant PR is waiting, do you need help getting this feature ready for merging and publishing?

**Comment by mymindstorm** [comment]
It's almost done, I'm just waiting a few days until translations come in for a menu string that was changed.

**Comment by ira-at-work** [comment]
well, I can help with Hebrew if it's missing... that's no reason to delay an important feature like that!

**Comment by Sneezry** [comment]
Hi @ira-bv, thanks for your interests to the contribution. You can help to translate at https://crowdin.com/project/authenticator-firefox

Hebrew has been translated 87%

![image](https://user-images.githubusercontent.com/1621293/110446409-18bab280-80fa-11eb-8279-a8491806168c.png)

**Comment by ira-at-work** [comment]
yup, it was at 64% when I started this morning, I think. all done now.

**Comment by ira-at-work** [comment]
I noticed I have 6.2.1 although the app store said 6.3.0 was out. I switched to developer mode to force an upgrade. Is that normal?

I still could not get the QR code imported (with a screenshot, not sure how you get it to the extension otherwise). I had to export from Google Auth, scan it with a QR reader (used Google lens, very irresponsible, but still possibly more trusted than other options unless you have one guaranteed to work offline, blocked from sending your scans to a third party), copied the decyphered text (the android clipboard, yet another risk) and pasted it in the extention's import screen, seems to have worked. had to remove a few duplicates of accounts I already had, but generally it worked OK.

Is there a plan to make this more streamlined or are we sticking to this?

TIA.

**Reply by mymindstorm** [reply]
> I noticed I have 6.2.1 although the app store said 6.3.0 was out. I switched to developer mode to force an upgrade. Is that normal?

Browsers may take a few days to receive the update, we have no control over this.

> Is there a plan to make this more streamlined or are we sticking to this?

There isn't much we can do to improve the QR scanning library we use. Normally a QR scan should just work.

**Reply by ira-at-work** [reply]
ok. is this on the way to the documentation of should I make a PR?

**Reply by mymindstorm** [reply]
Feel free to create a PR on Authenticator-Extension/authenticator.cc

**Comment by iamteedoh** [comment]
Just to clarify, here is how I got my Google Authenticator codes imported to this Authenticator:

- On my Android phone where Google Authenticator is installed, I selected **Transfer** to create the export. I did this one by one instead of choosing all at once just to be cautions I wouldn't lock myself out.
- Once the export was completed for the requested account, it'll show you a QR code on that phone for that account. Since the Google Authenticator doesn't allow taking screenshots of the QR code authenticator app, I grabbed a second phone and snapped a picture of the QR code
- The picture that I took, I then transferred it to my computer, where the Chrome browser with the Authenticator extension is installed
- I opened the picture on my Chrome browser where the extension is installed by pointing my browser to the location of the picture such as: Ctrl + o (windows / linux) or cmd + o (macos) and navigate to your file location
- Once the image was opened in my Chrome browser, I opened the Authenticator extension and selected Scan QR Code
- I then scanned the QR image that I opened on my browser and it then adds it automatically to the extension

That's it...now the codes are on your extension as well as your phone. Keep in mind this is the same process for Android or iPhone users.

**NOTE!!!** - Make sure you delete the QR codes you took a picture of from your phone and computer for security reasons!!! If you used iCloud/Photos or Google Drive/Phots to sync the picture to the cloud first before getting it on your computer, make sure you delete it from there so that it wipes it from all devices (phone/tablets, etc). However, you'll still need to manually remove the image from your computer as well. This is especially true if you don't have encryption enabled on your storage drive of said computer.

If someone gets a hold of these QR codes, they can add it to their own authenticator and now all they need it to brute force your password. BE CAUTIOUS !!!
